### PR TITLE
Add method to plot Precision/Recall curves 

### PR DESCRIPTION
**Pull request by warenlg**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -19,6 +19,7 @@ stages:
     if: tag =~ .*
 _install: &_install
   - pip install --upgrade pip cython codecov
+  - pip install matplotlib
   - pip install -e .[test]
```
Comment (vmarkovtsev) [inline .travis.yml]: `pip install -e .[test,plot]`

File: lookout/style/format/robustness.py [hunk 2]
```diff
@@ -187,45 +212,31 @@ def style_robustness_report(true_repo: str, noisy_repo: str, bblfsh: str, langua
...
-
     vnodes_y_true = files2vnodes(true_files, rules, client, language)
     mispreds_noise = files2mispreds(noisy_files, rules, client, language)
-
     diff_mispreds = get_diff_mispreds(mispreds_noise, lines_changed)
     changes_count = len(lines_changed)
-    print("Number of artificial mistakes potentially fixed by the model"
-          "(diff of mispredictions): %d / %d" % (len(diff_mispreds), changes_count))
-
+    log.info("Number of artificial mistakes potentially fixed by the model "
```
Comment (vmarkovtsev) [inline lookout/style/format/robustness.py]: log does `%` automatically:

```python
log.info("hellow %s %d", "world", 1)
```

**Comment by warenlg** [comment]
done thanks @vmarkovtsev

File: lookout/style/format/robustness.py [hunk 3]
```diff
@@ -187,45 +212,31 @@ def style_robustness_report(true_repo: str, noisy_repo: str, bblfsh: str, langua
...
+             "(diff of mispredictions): %d / %d", len(diff_mispreds), changes_count)
     style_fixes = get_style_fixes(diff_mispreds, vnodes_y_true, true_files, noisy_files)
+    log.info("style-analyzer fixes in the noisy repos: %d / %d -> %.1f %%"
```
Comment (vmarkovtsev) [inline lookout/style/format/robustness.py]: Missed here

Comment (warenlg) [reply lookout/style/format/robustness.py]: oups yep

File: lookout/style/format/__main__.py [hunk 4]
```diff
@@ -6,7 +6,7 @@
 from lookout.core.cmdline import ArgumentDefaultsHelpFormatterNoNone
 from lookout.core.slogging import setup as setup_slogging
 from lookout.style.format.quality_report import quality_report
-from lookout.style.format.robustness import style_robustness_report
+from lookout.style.format.robustness import style_robustness_report, plot_pr_curve
```
Comment (m09) [inline lookout/style/format/__main__.py]: wrong order of import. It's actually super strange that travis passes the style check considering that. Same thing in the test file.

**Comment by vmarkovtsev** [comment]
@m09 We need your review here

**Comment by vmarkovtsev** [comment]
```
...
======================================================================
ERROR: test_plot_pr_curve (lookout.style.format.tests.test_robustness.RobustnessTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/src-d/style-analyzer/lookout/style/format/tests/test_robustness.py", line 57, in test_plot_pr_curve
    output=tmpf.name)
  File "/home/travis/build/src-d/style-analyzer/lookout/style/format/robustness.py", line 340, in plot_pr_curve
    plot_curve(numpy.asarray(recalls), numpy.asarray(precisions), output)
  File "/home/travis/build/src-d/style-analyzer/lookout/style/format/robustness.py", line 280, in plot_curve
    plt.figure(figsize=(15, 10))
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/matplotlib/pyplot.py", line 533, in figure
    **kwargs)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/matplotlib/backend_bases.py", line 161, in new_figure_manager
    return cls.new_figure_manager_given_figure(num, fig)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/matplotlib/backends/_backend_tk.py", line 1046, in new_figure_manager_given_figure
    window = Tk.Tk(className="matplotlib")
  File "/opt/python/3.6.3/lib/python3.6/tkinter/__init__.py", line 2017, in __init__
    self.tk = _tkinter.create(screenName, baseName, className, interactive, wantobjects, useTk, sync, use)
_tkinter.TclError: no display name and no $DISPLAY environment variable
```
Waren needs to set a proper headless matplotlib backend (e.g., "Agg").

**Comment by m09** [comment]
@vmarkovtsev LGTM on my part

**Pull request merged**